src/main/java/org/itsallcode/junit/sysextensions/SystemOutGuard.java

```diff
@@ -0,0 +1,146 @@
+package org.itsallcode.junit.sysextensions;
+
+/*-
+ * #%L
+ * JUnit5 System Extensions
+ * %%
+ * Copyright (C) 2016 - 2018 itsallcode.org
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * #L%
+ */
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.Executable;
+
+import org.itsallcode.io.CapturingOutputStream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+/**
+ * This class implements a JUnit extension that lets you capture data written to
+ * {@link System#out}. in
+ *
+ * <p>
+ * For this it provides the user with a special kind of {@link OutputStream},
+ * the {@link CapturingOutputStream}. If you add that stream as a parameter to a
+ * method annotated with either <code>@BeforeTest</code> or <code>@Test</code> ,
+ * the {@link SystemOutGuard} injects an instance of the
+ * {@link CapturingOutputStream}.
```

Comment:
Better inject an interface that only offers required methods instead of a complete OutputStream with methods that are not used/might break things.

Reply:
Yep. My spider sense already tingled there, but in the end I overlooked that part.